Add crosslink tool

bryan-aguilar opened the pull request:
**Description:** Adds the [crosslink](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/crosslink) tool to simplify the function of adding replace statements to intra-repository go modules. 


_Please delete paragraphs that you did not use before submitting._

bryan-aguilar commented:
Can this PR get a skip changelog label?

bogdandrutu commented on Makefile:
```diff
@@ -433,3 +434,10 @@ checklinks:
+# error message "failed to sync logger:  sync /dev/stderr: inappropriate ioctl for device"
+# is a known issue but does not affect function. 
+.PHONY: crosslink
+crosslink: install-tools
```
we don't require install-tools in other places (unless I forget), just try to be consistent.

bryan-aguilar replied:
I believe there is some inconsistency. Multimod requires `install-tools`. But I will remove it to match the majority.

bogdandrutu reviewed (commented):
Lets make this step part of github action "check" and if git diffs then fail.

bryan-aguilar commented:
Updated GHA and make directive.

I also added the `--prune` flag as default. Since this will be running on PR build it should continue to clean the modules clean. You can see the details of the prune command [here](https://github.com/open-telemetry/opentelemetry-go-build-tools/tree/main/crosslink#prune---prune---p).
